src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -479,8 +483,34 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 
 		};
 
+		struct Curves
+		{
+
+			Curves( const IECore::CompoundObject *attributes )
+			{
+				mode = attributeValue<string>( g_curvesModeAttributeName, attributes, "ribbon" );
+				minPixelWidth = attributeValue<float>( g_curvesMinPixelWidthAttributeName, attributes, 0.0f );
+			}
+
+			IECore::InternedString mode;
+			float minPixelWidth;
+
+			void hash( IECore::MurmurHash &h ) const
+			{
+				h.append( mode );
+				h.append( minPixelWidth );
+			}
+
+			void apply( AtNode *node ) const
+			{
+				AiNodeSetStr( node, "mode", mode.c_str() );
```

Comment:
Yeah, lets get all 3 supported before closing the ticket. I don't mind if it comes before or after the OSLCode though... I don't really imagine we'll get a release out today in any case, so we have a little time I think. @boberfly can always build a branch locally if you think his testing is important before a proper rollout.